hazelcast/listener.py

```diff
@@ -105,24 +107,27 @@ def deregister_listener(self, user_registration_id):
                 )
                 self._invocation_service.invoke(invocation)
 
-                def handler(f, connection=connection):
+                def handler(f, captured_connection=connection):
                     e = f.exception()
-                    if e:
-                        if isinstance(
-                            e, (HazelcastClientNotActiveError, IOError, TargetDisconnectedError)
-                        ):
-                            return
-
-                        _logger.warning(
-                            "Deregistration of listener with ID %s has failed for address %s",
-                            user_registration_id,
-                            connection.remote_address,
-                        )
+                    if not e:
+                        return
+
+                    if isinstance(
+                        e, (HazelcastClientNotActiveError, IOError, TargetDisconnectedError)
+                    ):
+                        return
+
+                    _logger.warning(
+                        "Deregistration of listener with ID %s has failed for address %s",
+                        user_registration_id,
+                        captured_connection.remote_address,
+                    )
```

Comment:
Java client also [logs the exception](https://github.com/hazelcast/hazelcast/blob/b6c6c270ddffb20ce7dec1954e1d75ac4e5866ab/hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java#L343) here.

Reply:
Thanks. I added the error message to the log message. Also, removed the unused import that I added erronously.